### PR TITLE
[catch2] Add with_prefix option

### PR DESCRIPTION
**Pull request by mmatrosov**
Specify library name and version:  **catch2/2.x.x**

There is a useful `CATCH_CONFIG_PREFIX_ALL` configuration macro in Catch2 that allows to define all macros with a `CATCH_` prefix, thus avoiding possible name collisions (see https://github.com/catchorg/Catch2/blob/v2.13.7/docs/configuration.md#prefixing-catch-macros). It seems like a good idea to have this definition available as a library option.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/catch2/2.x.x/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -9,16 +9,24 @@ conan_basic_setup()
...
+    endif()
   endif()
-endif()
-
+else()
+  add_executable(test_package 000-CatchMain.cpp 400-with-prefix.cpp)
+  target_link_libraries(test_package PRIVATE Catch2::Catch2)
 
+  if(WITH_MAIN)
+    add_executable(standalone 400-with-prefix.cpp)
```
Comment (madebr) [inline recipes/catch2/2.x.x/test_package/CMakeLists.txt:29]: I think you forgot to add `400-with-prefix.cpp`.

Comment (mmatrosov) [reply recipes/catch2/2.x.x/test_package/CMakeLists.txt:29]: Whoops, you are right, fixed.

**Review by mathbunnyru — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by SpaceIm** [comment]
This option doesn't make sense. It's a pure interface definition. There is not even such CMake option.

**Pull request merged**